src/main/scala/net/jcazevedo/moultingyaml/YamlValue.scala

```diff
@@ -17,12 +17,13 @@ sealed abstract class YamlValue {
 
   def print(flowStyle: FlowStyle = FlowStyle.DEFAULT,
             scalarStyle: ScalarStyle = ScalarStyle.DEFAULT,
-            lineBreak: LineBreak = LineBreak.DEFAULT) = {
-    val printer = new SnakeYamlPrinter(flowStyle, scalarStyle, lineBreak)
-    printer(this)
+            lineBreak: LineBreak = LineBreak.DEFAULT) : String = {
```

Comment:
Yes, I agree. I would use an implicit parameter instead of a default one, so someone could define an implicit YamlPrinter or pass it explicitly as they wish.